develop/app/Controllers/v2/ProductController.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+namespace App\Controllers\v2;
+
+use App\Controllers\BaseController;
+use CodeIgniter\API\ResponseTrait;
+use App\Models\v2\ProductModel;
+use App\Entities\v2\ProductEntity;
+
+class ProductController extends BaseController
+{
+    use ResponseTrait;
+
+    /**
+     * [GET] api/v2/product/
+     * get all product
+     *
+     * @return void
+     */
+    public function index()
+    {
+        $limit  = $this->request->getGet("limit") ?? 10;
+        $offset = $this->request->getGet("offset") ?? 0;
+        $search = $this->request->getGet("search") ?? "";
+        $isDesc = $this->request->getGet("isDesc") ?? "desc";
+
+        $productEntity = new ProductEntity();
+        $productModel  = new ProductModel();
+
+        $query = $productModel->orderBy("created_at", $isDesc ? "DESC" : "ASC");
+        if ($search !== "") {
+            $query->like("name", $search);
+        }
+        $dataCount = $query->countAllResults(false);
+        $product = $query->findAll($limit, $offset);
+
+        $data = [
+            "list"   => [],
+            "dataCount" => $dataCount
+        ];
+
+        if ($product) {
+            foreach ($product as $productEntity) {
+                $productData = [
+                    "p_key"       => $productEntity->p_key,
+                    "name"        => $productEntity->name,
+                    "price"       => $productEntity->price,
+                    "amount"      => $productEntity->amount,
+                    "createdAt"   => $productEntity->createdAt,
+                    "updatedAt"   => $productEntity->updatedAt
+                ];
+                $data["list"][] = $productData;
+            }
+        } else {
+            return $this->fail("product data not found", 404);
+        }
+
+
+        return $this->respond([
+            "status" => true,
+            "data"  => $data,
+            "msg" => "product index method successful."
+        ]);
+    }
+
+
+    /**
+     * [GET] api/v2/product/{p_key}
+     * get someone product by p_key
+     *
+     * @param integer $p_key
+     * @return void
+     */
+    public function show($p_key = null)
+    {
+        if (is_null($p_key)) {
+            return $this->fail("Incoming data(Product Key) not true", 400);
```

Comment:
This content of error respond should be `The Product key is required`.

Reply:
This chahge is done.